packages/edit-post/src/plugins/index.js

```diff
@@ -24,8 +24,10 @@ registerPlugin( 'edit-post', {
 						<>
 							<MenuItem
 								role="menuitem"
-								href={ addQueryArgs( 'edit.php', {
-									post_type: 'wp_block',
+								href={ addQueryArgs( 'site-editor.php', {
+									path: '/library',
+									categoryType: 'pattern',
+									categoryId: 'my-patterns',
```

Comment:
May need to check how this works for roles that can't access the site editor. Maybe a fallback can be added.